pkg/solver/controller.go

```diff
@@ -300,6 +304,14 @@ func (controller *SolverController) solve(ctx context.Context) error {
 	}
 	span.AddEvent("add_deals.done")
 
+	span.AddEvent("report_deal_metrics.start")
+	storedDeals, err := controller.store.GetDealsAll()
+	if err != nil {
+		return err
+	}
+	reportDealMetrics(ctx, controller.meter, storedDeals)
+	span.AddEvent("report_deal_metrics.done")
```

Comment:
Ooh good catch! It would not record the `report_deal_metrics.done` span event. Also realized I wasn't handling the `reportDealMetrics` error.

Made some updates and now it looks like:

https://github.com/Lilypad-Tech/lilypad/blob/5e522c52d6ec1b8d13dad4031a236af4dc2e88d3/pkg/solver/controller.go#L307-L319

Note that we don't return an error when `reportDealMetrics` fails, we just record it in the trace. It's also logged in the `reportDealMetrics` function body.